cmake/macros/macro_deal_ii_package_handle.cmake

```diff
@@ -56,102 +56,103 @@ MACRO(DEAL_II_PACKAGE_HANDLE _feature _var)
   # value. We need this for modern™ MPI detection where CMake's
   # FindMPI.cmake might only set MPI_FOUND to true and nothing else.
   #
+  SET(_clear_variables_list "")
   IF(NOT DEFINED ${_feature}_FOUND)
     SET(${_feature}_FOUND TRUE)
   ENDIF()
 
-  SET(_variable ${_var})
-  SET(${_feature}_${_variable} "")
+  SET(_suffix "")
   SET(_required TRUE)
-  SET(_fine TRUE)
-  SET(_fill_clear FALSE)
-  SET(_clear "")
 
-  CLEAR_FEATURE(${_feature})
+  #
+  # Clear temporary variables
+  #
+  FOREACH(_suffix ${DEAL_II_LIST_SUFFIXES} ${DEAL_II_STRING_SUFFIXES})
+    set(_temp_${_suffix} "")
+  ENDFOREACH()
 
   FOREACH(_arg ${ARGN})
-    IF(_arg MATCHES "^LIBRARIES(|_DEBUG|_RELEASE)$"
-       OR _arg MATCHES "^(|BUNDLED_|USER_)INCLUDE_DIRS$"
-       OR _arg MATCHES "^(|USER_)DEFINITIONS(|_DEBUG|_RELEASE)$"
-       OR _arg MATCHES "^CXX_FLAGS(|_DEBUG|_RELEASE)"
-       OR _arg MATCHES "^LINKER_FLAGS(|_DEBUG|_RELEASE)"
-       OR _arg MATCHES "^EXECUTABLE(|_DEBUG|_RELEASE)")
-
-      IF(_fine)
-        IF(_variable MATCHES "^CXX_FLAGS(|_DEBUG|_RELEASE)"
-           OR _variable MATCHES "^LINKER_FLAGS(|_DEBUG|_RELEASE)")
-          TO_STRING(${_feature}_${_variable} ${${_feature}_${_variable}})
-        ENDIF()
-        MESSAGE(STATUS "  ${_feature}_${_variable}: ${${_feature}_${_variable}}")
-      ENDIF()
-
+    IF(("${_arg}" IN_LIST DEAL_II_LIST_SUFFIXES) OR ("${_arg}" IN_LIST DEAL_II_STRING_SUFFIXES))
       #
-      # *Yay* a new keyword.
+      # We encountered a new keyword.
       #
-      SET(_variable ${_arg})
-      SET(${_feature}_${_variable} "")
-      SET(_required TRUE)
-      SET(_fine TRUE)
+      set(_suffix "${_arg}")
+      SET(_fill_clear FALSE)
 
     ELSEIF("${_arg}" STREQUAL "REQUIRED")
       SET(_required TRUE)
+
     ELSEIF("${_arg}" STREQUAL "OPTIONAL")
       SET(_required FALSE)
+
     ELSEIF(_arg MATCHES "^(optimized|debug|general)$"
-            AND "${_variable}" STREQUAL "LIBRARIES")
-      #
-      # Keywords are special...
-      #
-      LIST(APPEND ${_feature}_${_variable} ${_arg})
-    ELSEIF("${_arg}" STREQUAL "CLEAR")
+            AND "${_suffix}" STREQUAL "LIBRARIES")
+      LIST(APPEND _temp_${_suffix} ${_arg})
+
+    ELSEIF("${_arg}" STREQUAL "CLEAR") # FIXME
```

Comment:
?

Reply:
I forgot to finish refactoring. The variable `_fill_clear` is now gone and the `FIXME`s have been removed.